cfbs/core.py

```diff
@@ -329,6 +330,10 @@ def _add_using_url(
             assert url.startswith(("https://", "git://", "ssh://"))
             config_path, url_commit = clone_url_repo(url)
 
+        if "@" in url:
+            assert url.split("@")[1] == url_commit
+            url = url.split("@")[0]
```

Comment:
This needs to be a bit more careful. This is the code I wrote when I was originally adding support for modules from git repos:
```
    if "@" in repo_url and (repo_url.rindex("@") > repo_url.rindex(".")):
        # commit specified in the url
        repo_url, commit = repo_url.rsplit("@", 1)
```
The reason for the complexity is that `user@some-URL` can appear in the URLs, so splitting on `@` must be done more carefully.

Reply:
Thank you, fixed. See last 2 commits.